### ISSUE TITLE
Gradle process can't be terminated

### ISSUE THREAD

**Issue by Emily-Jiang** [OpenLiberty/ci.gradle#610]
In this documentation, it suggests `gradlew libertyRun --no-daemon` can be terminated by ctrl+C. As a matter fact, ctrl+c does not terminate the process. The way to terminate the process is to issue libertyStop or kill the pid. Since this is difficult to fix, maybe reword the doc to say so.
https://github.com/OpenLiberty/ci.gradle/blob/main/docs/libertyRun.md#libertyrun-task

> Gradle's current daemon design makes it difficult to use a run task, as a Ctrl-C would kill the daemon while simultaneously leaving the application running in the background. Therefore, we need --no-daemon so that the signal can be received and properly handled. This is not true. Basically --no-daemon

**Comment by mattbsox** [comment]
Looks like this is related to #187 and #238

**Comment by ericglau** [comment]
This is partially fixed with #611 
When running with defaults (daemon) and pressing Ctrl+C, server stops after 10 seconds.
When running with --no-daemon and then press Ctrl+C, server stops right away.

Only remaining issue is when build.gradle has the following and you use daemon, then the server keeps running after pressing Ctrl+C.  Workaround is to run `gradle libertyStop` to stop server or `gradle --stop` to stop daemon. 
```
liberty {
    server {
        embedded = true
    }
}
```

**Comment by venmanyarun** [comment]
For embeded = true, we cannot make sure that Ctrl+C stops the server

* The daemon JVM is designed to stay alive between builds for performance. When you press Ctrl+C, it only terminates the client connection, not the daemon JVM itself.
* When embedded = true, the Liberty server runs inside the daemon JVM as threads, not as a separate process. These threads continue running because the daemon JVM continues running.
* The shutdown hook registered by the plugin only executes when the JVM terminates. Since the daemon JVM doesn't terminate on Ctrl+C, the hook never fires.

I have added a warning message where users are immediately informed about the limitation when using embedded mode, as part of https://github.com/OpenLiberty/ci.gradle/pull/1084
